### PR TITLE
New features on product-rating-summary

### PR DESCRIPTION
**Pull request by khrizzcristian**
#### What problem is this solving?
I added new features on product-rating-summary. They are all active on the workspace:

1. Ability to display stars even if the product has no reviews
2. Ability to hide the total number of reviews in product-rating-summary block
3. Ability to add `Add Review` button under the stars in product-rating-summary block

[Workspace](https://longo--customsoft.myvtex.com/ceas-crystal-frost-ls-hthtsts-5484070/p)

### CODE REVIEW

File: messages/en.json [hunk 1]
```diff
@@ -94,5 +94,9 @@
   "store/reviews.list.sortOptions.highestRated": "Highest Rated",
   "store/reviews.list.sortOptions.lowestRated": "Lowest Rated",
   "store/reviews.list.anonymous": "Anonymous",
-  "store/reviews.list.pagination.textOf": "of"
+  "store/reviews.list.pagination.textOf": "of",
+  "store/reviews.list.graph.stars": "{value} stars",
```
Comment (klzns) [inline messages/en.json:98]: You can use pluralization feature of ICU format

```
{value, plural, =0{0 stars} =1{1 star} other{# stars}}
```

You can read more here:
https://format-message.github.io/icu-message-format-for-translators/

File: messages/es.json [hunk 2]
```diff
@@ -94,5 +94,9 @@
   "store/reviews.list.sortOptions.highestRated": "Calificación más alta",
   "store/reviews.list.sortOptions.lowestRated": "Calificación más baja",
   "store/reviews.list.anonymous": "Anónimo",
-  "store/reviews.list.pagination.textOf": "de"
+  "store/reviews.list.pagination.textOf": "de",
+  "store/reviews.list.graph.stars": "Extrellas",
```
Comment (klzns) [inline messages/es.json:98]: Missing variable value

File: messages/pt.json [hunk 3]
```diff
@@ -94,5 +94,9 @@
   "store/reviews.list.sortOptions.highestRated": "Classificação mais alta",
   "store/reviews.list.sortOptions.lowestRated": "Classificação mais baixa",
   "store/reviews.list.anonymous": "Anônimo",
-  "store/reviews.list.pagination.textOf": "de"
+  "store/reviews.list.pagination.textOf": "de",
+  "store/reviews.list.graph.stars": "Estrelas",
```
Comment (klzns) [inline messages/pt.json:98]: Same

File: react/RatingSummary.tsx [hunk 4]
```diff
@@ -11,7 +11,12 @@ import { NormalizedCacheObject } from 'apollo-cache-inmemory'
 import { withApollo } from 'react-apollo'
 import { ProductContext } from 'vtex.product-context'
 import { useCssHandles } from 'vtex.css-handles'
+import { FormattedMessage } from 'react-intl'
+import { Link, canUseDOM } from 'vtex.render-runtime'
+import path from 'ramda/es/path'
```
Comment (klzns) [inline react/RatingSummary.tsx:16]: Ramda is a global package in IO, you can use this format:

```suggestion
import { path } from 'ramda'
```

File: react/RatingSummary.tsx [hunk 5]
```diff
@@ -115,13 +181,53 @@ const RatingSummary: FunctionComponent<Props> = props => {
...
+
+      if (!sessionRespose || !sessionRespose.namespaces) {
+        return
+      }
+
+      const { namespaces } = sessionRespose
+      const storeUserId = path(
+        ['authentication', 'storeUserId', 'value'],
+        namespaces
+      )
```
Comment (klzns) [inline react/RatingSummary.tsx:210]: You can avoid ramda with optional chaining in TS files
```suggestion
      const storeUserId = namespaces?.authentication?.storeUserId?.value
```

File: react/Reviews.tsx [hunk 6]
```diff
@@ -477,9 +490,18 @@ const Reviews: FunctionComponent<InjectedIntlProps & Props> = props => {
       .then((response: ApolloQueryResult<ReviewsData>) => {
         const reviews = response.data.reviewsByProductId.data
         const { total } = response.data.reviewsByProductId.range
+        const graphArray = [0, 0, 0, 0, 0, 0]
+        graphArray[0] = total
+        if (reviews) {
+          // eslint-disable-next-line array-callback-return
+          reviews.map((review: Review) => {
```
Comment (klzns) [inline react/Reviews.tsx:497]:
```suggestion
          reviews.forEach((review: Review) => {
```

File: react/Reviews.tsx [hunk 7]
```diff
@@ -477,9 +490,18 @@ const Reviews: FunctionComponent<InjectedIntlProps & Props> = props => {
       .then((response: ApolloQueryResult<ReviewsData>) => {
         const reviews = response.data.reviewsByProductId.data
         const { total } = response.data.reviewsByProductId.range
+        const graphArray = [0, 0, 0, 0, 0, 0]
+        graphArray[0] = total
+        if (reviews) {
+          // eslint-disable-next-line array-callback-return
```
Comment (klzns) [inline react/Reviews.tsx:496]: You can remove this if you use forEach
```suggestion
```

File: react/Reviews.tsx [hunk 8]
```diff
@@ -533,6 +558,170 @@ const Reviews: FunctionComponent<InjectedIntlProps & Props> = props => {
           </Fragment>
         )}
       </div>
+      {state.settings.showGraph ? (
+        <div className={`${handles.graphContent} mv5`}>
```
Comment (klzns) [inline react/Reviews.tsx:562]: Create a separate component for the graph

File: CHANGELOG.md [hunk 9]
```diff
@@ -5,6 +5,13 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+New features: 
+1. Ability to display stars even if the product has no reviews
+2. Ability to hide the total number of reviews in product-rating-summary block
+3. Ability to add `Add Review` button under the stars in product-rating-summary block
+
+### Added
+- Display graph option in app settings
```
Comment (klzns) [inline CHANGELOG.md:14]:
```suggestion
### Added
- Display graph option in app settings
- New settings to display stars even if the product has no reviews
- New settings to hide the total number of reviews in product-rating-summary block
- New settings to add `Add Review` button under the stars in product-rating-summary block
```

**Review by klzns — CHANGES_REQUESTED** [review]

File: react/Reviews.tsx (hunk 8 above)
Comment (khrizzcristian) [reply react/Reviews.tsx]: Done :)
Link active on: https://longo--biasicom.myvtex.com/aspirator-cu-sac-samsung-vc07m25e0wr-2-5-l-700-w-tub-telescopic-accesorii-2-in-1-rosu/p

File: docs/README.md [hunk 10]
```diff
@@ -38,6 +38,12 @@ Once in the app's settings page, define the following settings:
 
 - **Number of open review accordions** - Checking this box allows you to set a specific number of review accordions (instead of all of them) to automatically open when the page is loaded, displaying all the review text. If the `Default all review accordions to open` setting is active, this option is ignored.
 
+- **Display graph** - Checking this box allows you to display the reviews graph on the product page.
+
+- **Display stars in product-rating-summary if there are no reviews** - Checking this box allows you to display empty stars if the product has no reviews
+- **Display total reviews number on product-rating-summary block** - Checking this box allows you to display the total number of product reviews.
+- **Display `Add review` button on product-rating-summary block** - Checking this box allows you to display an Add review button under the stars.
```
Comment (guerreirobeatriz) [inline docs/README.md:45]:
```suggestion
- **Display graph** - Checking this box allows you to display the reviews graph on the product page.

- **Display stars in `product-rating-summary` if there are no reviews** - Checking this box allows you to display empty stars even if the product still has no reviews.

- **Display total reviews number on `product-rating-summary` block** - Checking this box allows you to display the total number of product reviews.

- **Display `Add review` button on `product-rating-summary` block** - Checking this box allows you to display an `Add review` button under the stars.
```

**Review by guerreirobeatriz — CHANGES_REQUESTED** [review]

File: docs/README.md [hunk 11]
```diff
@@ -87,6 +95,14 @@ In order to apply CSS customizations in this and other blocks, follow the instru
...
 | `summaryContainer`        |
 | `writeReviewContainer`    |
+| `graphContent`            |
+| `graphContainer`          |
+| `graphText`               |
+| `graphTextLabel`          |
+| `graphBarContainer`       |
+| `graphBar`                |
+| `graphBarPercent`         |
+| `summaryButtonContainer`  |
```
Comment (guerreirobeatriz) [inline docs/README.md:105]: This should be added to the table following the alphabetical order!

Comment (guerreirobeatriz) [reply docs/README.md:105]: Sorry for not pointing this out in my last review 😢

**Review by guerreirobeatriz — CHANGES_REQUESTED** [review]

**Review by guerreirobeatriz — APPROVED** [review]

**Comment by khrizzcristian** [comment]
Can you review this pull request pls? We are in stand by right now because we can't implement this

**Comment by polishq** [comment]
> Can you review this pull request pls? We are in stand by right now because we can't implement this

There have been a few releases since this PR was opened. Please rebase your branch and I'll add my approval.

**Comment by khrizzcristian** [comment]
> > Can you review this pull request pls? We are in stand by right now because we can't implement this
> 
> There have been a few releases since this PR was opened. Please rebase your branch and I'll add my approval.

Done :)

**Comment by polishq** [comment]
> > > Can you review this pull request pls? We are in stand by right now because we can't implement this
> > 
> > 
> > There have been a few releases since this PR was opened. Please rebase your branch and I'll add my approval.
> 
> Done :)

Please rebase from master. Your `manifest.json` shows version 2.2.1, but current version is 2.3.1.

**Comment by khrizzcristian** [comment]
> > > > Can you review this pull request pls? We are in stand by right now because we can't implement this
> > > 
> > > 
> > > There have been a few releases since this PR was opened. Please rebase your branch and I'll add my approval.
> > 
> > 
> > Done :)
> 
> Please rebase from master. Your `manifest.json` shows version 2.2.1, but current version is 2.3.1.

Rebased

**Comment by polishq** [comment]
> > > > > Can you review this pull request pls? We are in stand by right now because we can't implement this
> > > > 
> > > > 
> > > > There have been a few releases since this PR was opened. Please rebase your branch and I'll add my approval.
> > > 
> > > 
> > > Done :)
> > 
> > 
> > Please rebase from master. Your `manifest.json` shows version 2.2.1, but current version is 2.3.1.
> 
> Rebased

@khrizzcristian It looks like you performed a merge rather than a rebase. Please redo the rebase using these commands:
git checkout master
git checkout pull
git checkout -
git rebase master
git push --force

I also see that the Danger CI bot is throwing a failure. Not sure what that's about but let's see if the rebase fixes it.

**Comment by khrizzcristian** [comment]
> > > > > > Can you review this pull request pls? We are in stand by right now because we can't implement this
> > > > > 
> > > > > 
> > > > > There have been a few releases since this PR was opened. Please rebase your branch and I'll add my approval.
> > > > 
> > > > 
> > > > Done :)
> > > 
> > > 
> > > Please rebase from master. Your `manifest.json` shows version 2.2.1, but current version is 2.3.1.
> > 
> > 
> > Rebased
> 
> @khrizzcristian It looks like you performed a merge rather than a rebase. Please redo the rebase using these commands:
> git checkout master
> git checkout pull
> git checkout -
> git rebase master
> git push --force
> 
> I also see that the Danger CI bot is throwing a failure. Not sure what that's about but let's see if the rebase fixes it.

It shows me that everything is up to date. Rebased, pushed, pulled, etc...
My manifest.json shows "version": "2.3.2"

**Comment by polishq** [comment]
> It shows me that everything is up to date. Rebased, pushed, pulled, etc...
> My manifest.json shows "version": "2.3.2"

It's a common mistake, I've done it myself in the past. In a true rebase the commits that are in master are not shown here on this PR. We need that to be resolved.

**Comment by khrizzcristian** [comment]
> > It shows me that everything is up to date. Rebased, pushed, pulled, etc...
> > My manifest.json shows "version": "2.3.2"
> 
> It's a common mistake, I've done it myself in the past. In a true rebase the commits that are in master are not shown here on this PR. We need that to be resolved.

Rebased again :)

**Review by polishq — APPROVED** [review]

File: messages/es.json [hunk 12]
```diff
@@ -94,5 +94,8 @@
   "store/reviews.list.sortOptions.highestRated": "Calificación más alta",
   "store/reviews.list.sortOptions.lowestRated": "Calificación más baja",
   "store/reviews.list.anonymous": "Anónimo",
-  "store/reviews.list.pagination.textOf": "de"
+  "store/reviews.list.pagination.textOf": "de",
+  "store/reviews.list.graph.stars": "{value, plural, =0{0 extrellas} =1{1 extrella} other{# extrellas}}",
```
Comment (klzns) [inline messages/es.json:98]:
```suggestion
  "store/reviews.list.graph.stars": "{value, plural, =0{0 estrellas} =1{1 estrella} other{# estrellas}}",
```

**Review by klzns — CHANGES_REQUESTED** [review]

**Review by klzns — APPROVED** [review]

**Pull request merged**